### PR TITLE
Bugfix: Allow comments before module docstring noqa codes

### LINKED ISSUE #417: noqa cannot be used in module docstrings
It is not possible to use `# noqa`, e.g. to ignore something within a module docstring

For example, consider this file:

```python
# noqa: D400
"""Fullstop at end missing 1

Foo bar documentation.
""" # noqa: D400

def func(): # noqa: D400
"""Fullstop at end missing 2

Foo bar documentation.
"""
```
Which would result in the following pydocstyle error:

    /home/raabf/Dokumente/module_doc_test.py:2 at module level:
        D400: First line should end with a period (not '1')

Error codes—such as the D400 in the exapmle—can be ignored by putting an `# noqa: D400` at the end of the line for which the docstring is intended, such as the function definition. However, this is not possible for module docstrings. I did not find a place wher the noqa can be put for module docstrings. Is there a way to use the noqa for module docstrings?

### PR DESCRIPTION
**Pull request by bsamseth**
#427 allowed for adding noqa comments to module docstrings. However, this was not respected when there where additional comments _before_ the `# noqa` comment. For example:

```python
# -*- coding: utf-8 -*-
# noqa: D400
"""No period at the end""" 
```
This still warns about `D400`. 

This PR fixes this, effectively by discarding leading comments. 
Closes #417

- [X] Add unit tests and integration tests where applicable.  
- [x] Add a line to the release notes (docs/release_notes.rst) under "Current Development Version".  
      Make sure to include the PR number after you open and get one.

### CODE REVIEW

File: src/pydocstyle/parser.py [hunk 1]
```diff
@@ -576,12 +576,17 @@ def parse_definition(self, class_):
...
-        if self.current.kind == tk.COMMENT:
-            if 'noqa: ' in self.current.value:
-                skipped_error_codes = ''.join(
-                     self.current.value.split('noqa: ')[1:])
-            elif self.current.value.startswith('# noqa'):
-                skipped_error_codes = 'all'
+        while self.current.kind in (tk.COMMENT, tk.NEWLINE, tk.NL):
+            if self.current.kind == tk.COMMENT:
+                if 'noqa: ' in self.current.value:
+                    skipped_error_codes = ''.join(
```
Comment (samj1912) [inline src/pydocstyle/parser.py]: Also, this code as it is now, doesn't break after encounting noqa. We should be breaking after this.

Comment (bsamseth) [reply src/pydocstyle/parser.py]: Updated in the most recent commit.

File: src/pydocstyle/parser.py [hunk 2]
```diff
@@ -576,12 +576,17 @@ def parse_definition(self, class_):
     def parse_skip_comment(self):
         """Parse a definition comment for noqa skips."""
         skipped_error_codes = ''
-        if self.current.kind == tk.COMMENT:
-            if 'noqa: ' in self.current.value:
-                skipped_error_codes = ''.join(
-                     self.current.value.split('noqa: ')[1:])
-            elif self.current.value.startswith('# noqa'):
-                skipped_error_codes = 'all'
+        while self.current.kind in (tk.COMMENT, tk.NEWLINE, tk.NL):
```
[the remark this answers is not in the document]
Comment (bsamseth) [reply src/pydocstyle/parser.py]: I've tried all kinds of empty files, files with only comments etc. Nothing unexpected occurs. My understanding is that if the file has only comments, this code won't be called anyway.

As for adding a test for a file with all comments, I don't see that fitting in with the current testing setup. This is because the test case files are expected to have an `expectation` variable defined. Can't do that in an empty file. If you want such a test, how would you go about implementing it?

[the remark this answers is not in the document]
Comment (samj1912) [reply src/pydocstyle/parser.py:579]: You can add an integration test https://github.com/PyCQA/pydocstyle/blob/master/src/tests/test_integration.py#L333

See above for example. You can add tests with an empty file and check that the output code is 0

**Comment by Cielquan** [comment]
@bsamseth Any updates on this? :)

**Comment by bsamseth** [comment]
> @bsamseth Any updates on this? :)

I just haven't had time to prioritize this myself. You're more than welcome to add the test if you have time, otherwise, I'll get to it at some point.

**Comment by Cielquan** [comment]
> > @bsamseth Any updates on this? :)
> 
> I just haven't had time to prioritize this myself. You're more than welcome to add the test if you have time, otherwise, I'll get to it at some point.

Ok i will look into it. But how would I deliver my changes to this PR? 

Do i need to fork your fork and then you can merge my change into your fork and then it will be delivered here?

**Comment by bsamseth** [comment]
> > > @bsamseth Any updates on this? :)
> > 
> > 
> > I just haven't had time to prioritize this myself. You're more than welcome to add the test if you have time, otherwise, I'll get to it at some point.
> 
> Ok i will look into it. But how would I deliver my changes to this PR?
> 
> Do i need to fork your fork and then you can merge my change into your fork and then it will be delivered here?

You could if you prefer. I added you as a collaborator to my fork, so you can push to that fork as soon as you accept the invite. Thanks for taking the time!

File: src/pydocstyle/parser.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Cielquan) [reply src/pydocstyle/parser.py:587]: I added two integration tests:

1. Test that a file with only comments does only cause a D100.
2. Test that a file with comments and docstring does not throw an error.

**Comment by Cielquan** [comment]
see above

**Comment by bsamseth** [comment]
:tada: It only us 6 months, but it is finally in! Looking forward to integrating pydocstyle in our codebase, finally. Thanks for your time reviewing @samj1912, and thanks @Cielquan for pitching in with the tests.
